bundles/org.openhab.binding.monopriceaudio/src/main/java/org/openhab/binding/monopriceaudio/internal/communication/AmplifierModel.java

```diff
@@ -256,33 +258,26 @@ private static MonopriceAudioZoneDTO getMonopriceZoneData(String newZoneData) {
         this.padNumbers = padNumbers;
         this.zoneIds = zoneIds;
 
-        int i = 1;
-        for (String zoneId : zoneIds) {
-            zoneIdMap.put(zoneId, "zone" + i);
-            i++;
-        }
+        // Build a map between the amp's physical zone IDs and the thing's logical zone names
+        final Map<String, String> zoneIdMap = new HashMap<>();
+        IntStream.range(0, zoneIds.size()).forEach(i -> zoneIdMap.put(zoneIds.get(i), "zone" + (i + 1)));
+        this.zoneIdMap = Collections.unmodifiableMap(zoneIdMap);
     }
 
     public abstract MonopriceAudioZoneDTO getZoneData(String newZoneData);
 
     public abstract List<StateOption> getSourceLabels(MonopriceAudioThingConfiguration config);
 
     public String getZoneIdFromZoneName(String zoneName) {
-        for (String zoneId : zoneIdMap.keySet()) {
-            if (zoneIdMap.get(zoneId).equals(zoneName)) {
-                return zoneId;
-            }
-        }
-        return "";
```

Comment:
How is this an improvement over a simple for loop, looking for some education here?

Reply:
The first commit was pretty slick in that it was done with a one-liner. However after the latest update of lastnpe, the build was failing because it was unhappy that Map.get() could return null. And I agree that the latest change to remedy that was pretty ugly, so I reverted back to the original for loops.